werkit/aws_lambda/parallel.py

```diff
@@ -0,0 +1,46 @@
+import pdb
+import aioboto3
+import json
+import asyncio
+import os
+from functools import partial
+from botocore.exceptions import ClientError
+import time
+from botocore.client import Config
+
+
+async def call_worker_service(lambda_worker_function_name, extra_args, _input):
+    async with aioboto3.client("lambda") as lambda_client:
+        # tic = time.time()
+        response = await lambda_client.invoke(
+            FunctionName=lambda_worker_function_name,
+            Payload=json.dumps({"input": _input, "extra_args": extra_args}),
+        )
+        # toc = time.time()
+        # print(_input, toc - tic)
+        payload = await response["Payload"].read()
+        return json.loads(payload.decode())
+
+
+async def wait_for(timeout, lambda_worker_function_name, extra_args, _input):
+    try:
+        return await asyncio.wait_for(
+            call_worker_service(lambda_worker_function_name, extra_args, _input),
+            timeout=timeout,
+        )
+    except ClientError as ex:
+        return ex
+    except asyncio.TimeoutError as ex:
+        return ex
+    except Exception as ex:
+        return ex
+
+
+async def parallel_map_on_lambda(
+    lambda_worker_function_name, timeout, input, extra_args=[]
+):
```

Comment:
Could you add a sentence of doc to these? At least to the one (I think it's this one, right?) that could be used without the orchestrator lambda.